Info-Orbs/src/widgets/webDataWidget.cpp

```diff
@@ -0,0 +1,158 @@
+}
+
+void WebDataWidget::draw(bool force) {
+    uint32_t background = TFT_BLACK;
+    uint32_t color = TFT_CYAN;
+
+    // Weather, displays a clock, city & text weather discription, weather icon, temp, 3 day forecast
+    for (int i=0 ; i<5; i++) {
+        String fingerprint = getFingerprint(i);
+        if (fingerprint != m_fingerprint[i]) {
+            m_manager.selectScreen(i);
+
+            TFT_eSPI &display = m_manager.getDisplay();
+            display.fillScreen(m_background[i]);
+            display.setTextColor(m_labelColor[i]);
+            display.setTextSize(2);
+            display.setTextDatum(MC_DATUM);
+            display.drawString(m_label[i], 120, 75, 2);
+            display.setTextColor(m_color[i]);
+            display.setTextDatum(MC_DATUM);
+            if (m_data[i].length()>10) {
+                display.setTextSize(2);
+            } else {
+                display.setTextSize(3);
+            }
+            display.drawString(m_data[i], 120, 120, 2);
+            m_fingerprint[i] = fingerprint;
+        }
+
+    }
+}
+
+void WebDataWidget::update() {
+    update(false);
+}
+
+void WebDataWidget::update(bool force) {
+    if (force || m_lastUpdate == 0 || (millis() - m_lastUpdate) >= m_updateDelay) {
+        HTTPClient http;
+        http.begin(httpRequestAddress);
+        int httpCode = http.GET();
+
+        if (httpCode > 0) {  // Check for the returning code
+            JsonDocument doc;
+            DeserializationError error = deserializeJson(doc, http.getString());
+            if (!error) {
+                for(int8_t i=0;i<5;i++) {
+                    m_label[i] = doc[i]["label"].as<String>();
+                    m_data[i] = doc[i]["data"].as<String>();
+                    if (doc[i].containsKey("color")) {
+                        m_color[i] = stringToColor(doc[i]["color"].as<String>());
+                    } else {
+                        m_color[i] = m_defaultColor;
+                    }
+                    if (doc[i].containsKey("labelColor")) {
+                        m_labelColor[i] = stringToColor(doc[i]["labelColor"].as<String>());
+                    } else {
+                        m_labelColor[i] = m_color[i];
+                    }
+                    if (doc[i].containsKey("background")) {
+                    m_background[i] = stringToColor(doc[i]["background"].as<String>());
+                    } else {
+                        m_background[i] = m_defaultBackground;
+                    }
+                }
+                
+                m_lastUpdate = millis();
+            } else {
+                // Handle JSON deserialization error
+                Serial.println("deserializeJson() failed");
+            }
+        } else {
+            // Handle HTTP request error
+            Serial.printf("HTTP request failed, error: %s\n", http.errorToString(httpCode).c_str());
+        }
+        http.end();
+    }
+}
+
+int32_t WebDataWidget::stringToColor(String color) {
```

Comment:
This should probably be in a global library as it seems like it will be useful for multiple widgets.